### PR TITLE
chore(Link): Remove the CSS modules feature flag from the Link component

### PR DESCRIPTION
**Pull request by jonrohan**
#### Changed

Remove the CSS modules feature flag from the Link component

### Rollout strategy

- [ ] Patch release
- [x] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Testing & Reviewing

Integration tests in dotcom

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/Link/Link.tsx [hunk 1]
```diff
@@ -121,11 +60,11 @@ const Link = forwardRef(({as: Component = 'a', className, inline, underline, ...
...
-      className={className}
+    <Component
+      className={clsx(className, classes.Link)}
+      data-muted={props.muted}
       data-inline={inline}
-      underline={underline}
+      data-underline={underline}
       {...props}
       // @ts-ignore shh
       ref={innerRef}
```
Comment (jonrohan) [inline packages/react/src/Link/Link.tsx:70]: I probably don't need this ref anymore? @joshblack

File: packages/react/src/Link/Link.tsx [hunk 2]
```diff
@@ -1,18 +1,14 @@
...
-import sx from '../sx'
 import classes from './Link.module.css'
-import {useFeatureFlag} from '../FeatureFlags'
 import Box from '../Box'
 import type {ComponentProps} from '../utils/types'
 import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
 
 type StyledLinkProps = {
+  /** @deprecated use CSS modules to style hover color */
   hoverColor?: string
```
Comment (joshblack) [inline packages/react/src/Link/Link.tsx:12]: If you haven't already, could you add a note to https://github.com/github/primer/issues/3541 for removing this from `Link`? 🙏

File: packages/react/src/Link/Link.tsx (hunk 1 above)
Comment (joshblack) [reply packages/react/src/Link/Link.tsx:71]: I think it's used in the effect for checking the ref type IIRC, let me know if I'm missing something 👀

File: packages/react/src/Link/Link.tsx [hunk 3]
```diff
@@ -21,49 +17,7 @@ type StyledLinkProps = {
...
-
-  /* Inline links (inside a text block), however, should have underline based on accessibility setting set in data-attribute */
-  /* Note: setting underline={false} does not override this */
-  [data-a11y-link-underlines='true'] &[data-inline='true'] {
-    text-decoration: underline;
-  }
-
-  &:hover {
-    text-decoration: ${props => (props.muted ? 'none' : 'underline')};
-    ${props => (props.hoverColor ? hoverColor : props.muted ? `color: ${get('colors.accent.fg')(props)}` : '')};
```
Comment (joshblack) [inline packages/react/src/Link/Link.tsx:48]: Are we passing along the `hoverColor` prop now that it's been decorated? I wasn't sure if we were still supporting it or not and wanted to double check

Comment (jonrohan) [reply packages/react/src/Link/Link.tsx:48]: It's not doing anything anymore

**Review by joshblack — APPROVED** [review]

**Pull request merged**